### PR TITLE
feat: Added component query helpers for expected result size

### PR DESCRIPTION
**Pull request by mcollovati**
Added helpers to verify the size of the search results:
exact size, in a range, min and max

Part of #1370

### CODE REVIEW

File: vaadin-testbench-core/src/test/java/com/vaadin/testbench/unit/ComponentQueryTest.java [hunk 1]
```diff
@@ -475,6 +475,246 @@ void withCondition_predicateMatched_getsComponents() {
...
+        Assertions.assertThrows(IllegalArgumentException.class,
+                () -> query.withResultsSize(-1));
+    }
+
+    @Test
+    void withResultsSize_afterWithMaxResults_overwritesRange() {
+        ComponentQuery<Div> query = $(Div.class);
+        Assertions.assertThrows(IllegalArgumentException.class,
+                () -> query.withResultsSize(-1));
+    }
```
Comment (caalador) [inline vaadin-testbench-core/src/test/java/com/vaadin/testbench/unit/ComponentQueryTest.java:532]: This test now tests negative throws.

File: vaadin-testbench-core/src/test/java/com/vaadin/testbench/unit/ComponentQueryTest.java [hunk 2]
```diff
@@ -475,6 +475,246 @@ void withCondition_predicateMatched_getsComponents() {
...
+
+    @Test
+    void withResultsSize_afterWithMinResults_overwritesRange() {
+        ComponentQuery<Div> query = $(Div.class);
+        Assertions.assertThrows(IllegalArgumentException.class,
+                () -> query.withResultsSize(-1));
+    }
```
Comment (caalador) [inline vaadin-testbench-core/src/test/java/com/vaadin/testbench/unit/ComponentQueryTest.java:539]: This test also tests negative throws

File: vaadin-testbench-core/src/test/java/com/vaadin/testbench/unit/ComponentQueryTest.java [hunk 3]
```diff
@@ -475,6 +475,246 @@ void withCondition_predicateMatched_getsComponents() {
...
+
+    @Test
+    void withResultsSize_afterWithResults_overwritesRange() {
+        ComponentQuery<Div> query = $(Div.class);
+        Assertions.assertThrows(IllegalArgumentException.class,
+                () -> query.withResultsSize(-1));
+    }
```
Comment (caalador) [inline vaadin-testbench-core/src/test/java/com/vaadin/testbench/unit/ComponentQueryTest.java:546]: This test also tests negative throws

File: vaadin-testbench-core/src/test/java/com/vaadin/testbench/unit/ComponentQueryTest.java (hunk 1 above)
Comment (mcollovati) [reply vaadin-testbench-core/src/test/java/com/vaadin/testbench/unit/ComponentQueryTest.java]: Great catch! Thanks
fixed

File: vaadin-testbench-core/src/test/java/com/vaadin/testbench/unit/ComponentQueryTest.java (hunk 2 above)
Comment (mcollovati) [reply vaadin-testbench-core/src/test/java/com/vaadin/testbench/unit/ComponentQueryTest.java]: fixed

**Review by caalador — APPROVED** [review]

**Comment by caalador** [comment]
Has conflicts.

**Comment by mcollovati** [comment]
rebased again
